src/utils/ucc_math.h

```diff
@@ -21,4 +22,26 @@ static inline size_t ucc_dt_size(ucc_datatype_t dt)
     return 0;
 }
 
+static inline size_t
+ucc_coll_args_get_count(const ucc_coll_args_t *args, const ucc_count_t *counts,
+                        ucc_rank_t idx)
+{
+    if ((args->mask & UCC_COLL_ARGS_FIELD_FLAGS) &&
+        (args->flags & UCC_COLL_ARGS_FLAG_COUNT_64BIT)) {
+        return ((uint64_t *)counts)[idx];
+    }
+    return ((uint32_t *)counts)[idx];
+}
+
+static inline size_t
+ucc_coll_args_get_displacement(const ucc_coll_args_t *args,
+                               const ucc_aint_t *displacements, ucc_rank_t idx)
+{
+    if ((args->mask & UCC_COLL_ARGS_FIELD_FLAGS) &&
+        (args->flags & UCC_COLL_ARGS_FLAG_DISPLACEMENTS_64BIT)) {
+        return ((uint64_t *)displacements)[idx];
+    }
+    return ((uint32_t *)displacements)[idx];
+}
+
```

Comment:
IMO, these are helper functions of core collectives. It seems not appropriate to have these func to define in generic math utils.

Reply:
Do you mean move it from utils/math to utils/coll_utils?

Reply:
👍